consensus/polybft/validator_set.go

```diff
@@ -18,46 +86,430 @@ type ValidatorSet interface {
 
 	// Accounts returns the list of the ValidatorMetadata
 	Accounts() AccountSet
+
+	// IncrementProposerPriority increments priorities number of times
+	IncrementProposerPriority(times uint64) error
 }
 
 type validatorSet struct {
 	// last proposer of a block
 	last types.Address
 
-	// current list of validators (slice of (Address, BlsPublicKey) pairs)
-	validators AccountSet
+	// validators represents current list of validators with their priority
+	validators []*ValidatorAccount
+
+	// proposer of a block
+	proposer *ValidatorAccount
+
+	// totalVotingPower denotes voting power of entire validator set
+	totalVotingPower int64
 }
 
-func newValidatorSet(lastProposer types.Address, validators AccountSet) *validatorSet {
-	return &validatorSet{validators: validators, last: lastProposer}
+// NewValidatorSet creates a new validator set.
+func NewValidatorSet(valz AccountSet) (*validatorSet, error) {
+	var validators = make([]*ValidatorAccount, len(valz))
+	for i, v := range valz {
+		validators[i] = NewValidator(v)
+	}
+
+	validatorSet := &validatorSet{
+		validators: validators,
+		// TODO voting power
+		// votingPowerMap: make(map[pbft.NodeID]uint64, len(validators)),
```

Comment:
Should we merge something first?

Reply:
@b00ris that votingPoweMap is fixed here: https://github.com/0xPolygon/polygon-edge/pull/904